cli/fathom_web/test/test_train.py

```diff
@@ -32,3 +33,146 @@ def test_auto_vectorization_smoke(tmp_path):
+    # Partial reduction in number of cutoffs
+    y_pred = tensor([-2.1605, -0.5696, 0.4886, 0.8633, -1.3479,
+                     -0.5813, -0.5696, 0.5696, -0.5950, -0.5696])
+    expected = [0.15, 0.28, 0.36, 0.49, 0.63, 0.67]
+    possibles = possible_cutoffs(y_pred)
+    assert len(possibles) == 6
+    assert all([a == b for a, b in zip(possibles, expected)])
+
+    # No reduction in number of cutoffs
+    y_pred = tensor([-2, -2.25, -1.95, 1.251])
+    expected = [0.11, 0.12, 0.45]
+    possibles = possible_cutoffs(y_pred)
+    assert len(possibles) == 3
+    assert all([a == b for a, b in zip(possibles, expected)])
+
+
+def test_find_optimal_cutoff_single_cutoff_with_highest_accuracy():
+    # This test is doing the steps completed by find_optimal_cutoff separately to
+    # determine the expected value.  The functions used are covered by other tests.
+    y_pred = tensor([-2.1605, -0.5696, 0.4886, 0.8633, -1.3479, -0.5813, -0.5696, 0.5696, -0.5950, -0.5696])
+    y = tensor([0., 0., 1., 1., 0., 0., 0., 1., 0., 0.])
+
+    # Determining the expected_cutoff
+    possibles = possible_cutoffs(y_pred)  # this list will contain the optimal cutoff
+
+    # Get the accuracy for each possible cutoff
+    # Note this is a different method of tracking the best cutoff than in find_optimal_cutoff.
+    cutoff_accuracy = {}
+    for possible in possibles:
+        accuracy, _, _ = accuracy_per_tag(y, y_pred, possible, num_prunes=0)
+        cutoff_accuracy[possible] = accuracy
+
+    # Get the key/value that has the highest accuracy.
+    expected_cutoff = max(cutoff_accuracy.items(), key=operator.itemgetter(1))[0]
+    max_accuracy = max(cutoff_accuracy.items(), key=operator.itemgetter(1))[1]
+
+    # Verify that there is only 1 cutoff with this max accuacy.
+    accuracy_occurrences = 0
+    for accuracy in cutoff_accuracy.values():
+        if max_accuracy == accuracy:
+            accuracy_occurrences += 1
+    assert accuracy_occurrences == 1
+
+    # Now that we have the expected accuracy check the value returned from
+    # find_optimal_cutoff against it (this is the real test)
+    optimal_cutoff = find_optimal_cutoff(y, y_pred, num_prunes=0)
+    assert optimal_cutoff == expected_cutoff
+    # and a final double check
+    assert optimal_cutoff == 0.49
+
+
+def test_find_optimal_cutoff_multiple_cutoffs_with_highest_accuracy():
+    # This test is doing the steps completed by find_optimal_cutoff separately to
+    # determine the expected value.  The functions used are covered by other tests.
+    y_pred = tensor([-2, -1.5, -1, -0.5, 0, 0.5, 1, 1.25, 2, 2.5])
+    y = tensor([0., 1., 0., 1., 0., 1., 0., 1., 0., 1.])
+
+    # Determining the expected_cutoff
+    possibles = possible_cutoffs(y_pred)  # this list will contain the optimal cutoff
+    assert len(possibles) == 9
+
+    # get the accuracy for each possible cutoff
+    # Note this is a different method of tracking the best cutoff than in find_optimal_cutoff.
+    cutoff_accuracy = {}
```

Comment:
I'd rather not have this loop (and maybe the next one too) repeated across 2 tests; it's a nontrivial amount of code, and factoring it out would make it both quicker to understand the test suite and less prone to bugs and pointless accumulation of differences.